lib/task-service.js

```diff
@@ -4,21 +4,54 @@
 
 var di = require('di');
 
-module.exports = factory;
-di.annotate(factory, new di.Provide('Services.Task'));
-di.annotate(factory, new di.Inject(
+module.exports = taskServiceFactory;
+di.annotate(taskServiceFactory, new di.Provide('Services.Task'));
+di.annotate(taskServiceFactory, new di.Inject(
+    'Logger',
+    'Promise',
+    'ChildProcess',
     'TaskOption.Validator'
 ));
 
-function factory(validator) {
+function taskServiceFactory(
+    Logger,
+    Promise,
+    ChildProcess,
+    validator
+) {
+    var logger = Logger.initialize(taskServiceFactory);
+    var requiredCmdTools = ['ipmitool'];
+
     function TaskService() {}
 
     TaskService.prototype.start = function() {
-        return validator.register();
+        var self = this;
+        return Promise.resolve()
```

Comment:
Actually, that's designed purposely. As I see some function returns Promise, such as `validator.register` and `checkToolsExist`, but some function doesn't return Promise, such as `validator.reset`. If I directly chain promise after the function, I need to take special care for the return value of each promise. But if I put these functions into nesting promise,s I don't need to care their return values.

Actually  this doesn't bring too much benefit since only a few sub-functions in `start`/`stop` at present. It brings flexibility only if more and more sub-functions are put here in future.

So I'm OK to change this or not. If you insist on the Promise.resolve() is redundant, I'm OK to remove it.